### PR TITLE
DOC: Code from markevery_prop_cycle moved to test.

### LINKED ISSUE #21863: [Doc]: Remove example "prop_cycle property markevery in rcParams"
### Documentation Link

https://matplotlib.org/devdocs/gallery/lines_bars_and_markers/markevery_prop_cycle.html

### Problem

This is basically a proof that a bug was fixed, but is not instructive as an example (and horrible from a visual perspective).

### Suggested improvement

Convert the code into a test.

A `@check_figures_equal` comparing rcParams prop-cycle vs. explicitly set via keyword should do.

### Matplotlib Version

3.5.0

### PR DESCRIPTION
**Pull request by ojeda-e**
## PR Summary
Fixes #21863
Code from [markevery_prop_cycle](https://matplotlib.org/devdocs/gallery/lines_bars_and_markers/markevery_prop_cycle.html) moved to unit test in `test_lines.py`. The new unit test, `test_markevery_prop_cycle`, uses `@check_figures_equal` as suggested by @timhoffm.

To make the unit test more readable and shorter I added.
- Fixture including the cases.
- Color list from colormap.

Thanks!

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/tests/test_lines.py [hunk 1]
```diff
@@ -343,3 +352,25 @@ def test_input_copy(fig_test, fig_ref):
...
+    axs = fig_ref.add_axes([0.1, 0.1, 0.6, 0.75])
+    for i in range(len(cases)):
+        axs.plot(yy[:, i], markevery=cases[i],  c=colors[i])
+
+    matplotlib.rcParams['axes.prop_cycle'] = cycler(markevery=cases,
+                                                    color=colors)
+
+    ax = fig_test.add_axes([0.1, 0.1, 0.6, 0.75])
+    for i in range(len(cases)):
+        ax.plot(yy[:, i])
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_lines.py:376]:
```suggestion
    ax = fig_test.add_subplot()
    for i, _ in enumerate(cases):
        ax.plot(y - i, 'o-')
```

**Review by timhoffm — COMMENTED** [review]
Thanks for the PR! I've added some suggestions.

File: lib/matplotlib/tests/test_lines.py [hunk 2]
```diff
@@ -343,3 +352,25 @@ def test_input_copy(fig_test, fig_ref):
     # Trigger cache invalidation
     l.set_drawstyle("steps")
     fig_ref.add_subplot().plot([0, 2, 4], [0, 2, 4], ".-", drawstyle="steps")
+
+
+@check_figures_equal(extensions=["png"])
+def test_markevery_prop_cycle(fig_test, fig_ref, cases):
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_lines.py:358]:
```suggestion
def test_markevery_prop_cycle(fig_test, fig_ref, cases):
    """Test that we can set markevery prop_cycle."""
```
To make it more clear to the reader what the test does.

File: lib/matplotlib/tests/test_lines.py [hunk 3]
```diff
@@ -20,6 +20,15 @@
 from matplotlib.testing.decorators import image_comparison, check_figures_equal
 
 
+@pytest.fixture
+def cases():
+    """Define a list of markevery cases to plot."""
+    cases = [None, 8, (30, 8), [16, 24, 30], [0, -1],
+             slice(100, 200, 3), 0.1, 0.3, 1.5,
+             (0.0, 0.1), (0.45, 0.1)]
+    return cases
```
[the remark this answers is not in the document]
Comment (ojeda-e) [reply lib/matplotlib/tests/test_lines.py:29]: Gotcha, thanks for clarifying! I'll move it to the unit test.

File: lib/matplotlib/tests/test_lines.py (hunk 2 above)
Comment (ojeda-e) [reply lib/matplotlib/tests/test_lines.py:358]: Thanks, added!

File: lib/matplotlib/tests/test_lines.py [hunk 4]
```diff
@@ -343,3 +352,25 @@ def test_input_copy(fig_test, fig_ref):
...
+    cmap = plt.get_cmap('jet')
+    colors = cmap(np.linspace(0.2, 0.8, len(cases)))
+
+    x = np.linspace(0, 2 * np.pi)
+    offsets = np.linspace(0, 2 * np.pi, 11, endpoint=False)
+    yy = np.transpose([np.sin(x + phi) for phi in offsets])
+
+    axs = fig_ref.add_axes([0.1, 0.1, 0.6, 0.75])
+    for i in range(len(cases)):
+        axs.plot(yy[:, i], markevery=cases[i],  c=colors[i])
```
[the remark this answers is not in the document]
Comment (ojeda-e) [reply lib/matplotlib/tests/test_lines.py:369]: Ouch, thanks for catching this.

**Review by timhoffm — APPROVED** [review]
Thanks, this looks good!

One final thing, please delete `examples/lines_bars_and_markers/markevery_prop_cycle.py`.

**Comment by tacaswell** [comment]
We reference the example from the whats new

```
/home/circleci/project/doc/users/prev_whats_new/whats_new_3.0.rst:110: WARNING: unknown document: /gallery/lines_bars_and_markers/markevery_prop_cycle
```

I suspect the best solution here is to edit the (old) whats new to remove the reference.

**Comment by ojeda-e** [comment]
Thanks @tacaswell, deleted the reference in `/doc/users/prev_whats_new/whats_new_3.0.rst`.

**Comment by timhoffm** [comment]
Thanks @ojeda-e, and congratulations on your first contribution to Matplotlib! We'd be happy to see you back!

**Pull request merged**